miio/viomivacuum.py

```diff
@@ -268,124 +380,398 @@ def has_map(self) -> bool:
 
     @property
     def has_new_map(self) -> bool:
-        """TODO: unknown"""
+        """True if the device has scanned a new map (like a new floor)."""
         return bool(self.data["has_newmap"])
 
     @property
     def mop_mode(self) -> ViomiMode:
-        """Whether mopping is enabled and if so which mode.
+        """Whether mopping is enabled and if so which mode."""
+        return ViomiMode(self.data["is_mop"])
 
-        TODO: is this really the same as mode?
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self.data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self.data["hw_info"]
+
+    @property
+    def charging(self) -> bool:
+        """True if battery is charging.
+
+        Note: When the battery is at 100%, device reports that it is not charging.
         """
-        return ViomiMode(self.data["is_mop"])
+        return not bool(self.data["is_charge"])
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is working."""
+        return not bool(self.data["is_work"])
+
+    @property
+    def light_state(self) -> bool:
+        """Led state.
+
+        This seems doing nothing on STYJ02YM
+        """
+        return bool(self.data["light_state"])
+
+    @property
+    def map_number(self) -> int:
+        """Number of saved maps."""
+        return self.data["map_num"]
+
+    @property
+    def mop_route(self) -> ViomiRoutePattern:
+        """Pattern mode."""
+        return ViomiRoutePattern(self.data["mop_route"])
+
+    @property
+    def order_time(self) -> int:
+        """FIXME: ??? int or bool."""
+        return self.data["order_time"]
```

Comment:
Do you know what does this do? If not, it's probably better to remove it for now and re-add it later on to avoid breaking the API in the future. Same applies to other properties marked with `FIXME` below, so I'm not going to comment them separately.

Reply:
fixed